spring-cloud-dataflow-tasklauncher/spring-cloud-dataflow-tasklauncher-sink/src/test/java/org/springframework/cloud/dataflow/tasklauncher/sink/TaskLauncherSinkTests.java

```diff
@@ -66,244 +56,140 @@
+		resource.setType("default");
+		resource.setName("default");
+		TaskOperations taskOperations = context.getBean(TaskOperations.class);
+		when(taskOperations.currentTaskExecutions()).thenReturn(Collections.singletonList(resource));
+
+		LaunchRequest launchRequest = new LaunchRequest("test", Collections.emptyList(), Collections.emptyMap());
+		InputDestination inputDestination = context.getBean(InputDestination.class);
+		logger.info("sending:input={}", launchRequest);
+		long start = System.currentTimeMillis();
+		inputDestination.send(new GenericMessage<>(launchRequest));
+		Awaitility.await("Error produced").until(errorReceived::get);
+		long total = System.currentTimeMillis() - start;
+		assertThat(total).isGreaterThan(600L);
 	}
 
 	@Test
-	public void backoffWhenNoMessages() {
-
-		contextRunner.withPropertyValues(
-						"trigger.period=10",
-						"trigger.initial-delay=0",
-						"messageSourceDisabled=true",
-						"countDown=3")
-				.run(context -> {
-					CountDownLatch countDownLatch = countDownLatch(context);
-					CurrentTaskExecutionsResource currentTaskExecutionsResource = currentTaskExecutionsResource(
-							context);
-					DynamicPeriodicTrigger trigger = trigger(context);
-
-					assertThat(countDownLatch.await(1, TimeUnit.SECONDS)).isTrue();
-					assertThat(currentTaskExecutionsResource.getRunningExecutionCount()).isZero();
-					assertThat(trigger.getDuration()).isGreaterThanOrEqualTo(Duration.ofMillis(40));
-				});
+	public void launchValidRequest() {
+		AtomicBoolean errorReceived = new AtomicBoolean();
+		errorReceived.set(false);
+		SubscribableChannel errorChannel = context.getBean("errorChannel", SubscribableChannel.class);
+		errorChannel.subscribe(message -> {
+			try {
+				logger.info("received:error:{}", message);
+				errorReceived.set(true);
+			}
+			catch (Exception e) {
+				fail(e.toString());
+			}
+		});
+		CurrentTaskExecutionsResource resource = new CurrentTaskExecutionsResource();
+		resource.setRunningExecutionCount(0);
+		resource.setMaximumTaskExecutions(8);
+		resource.setType("default");
+		resource.setName("default");
+		TaskOperations taskOperations = context.getBean(TaskOperations.class);
+		when(taskOperations.currentTaskExecutions()).thenReturn(Collections.singletonList(resource));
+		when(taskOperations.launch(anyString(), anyMap(), anyList()))
+			.thenReturn(new LaunchResponseResource(1, "boot3"));
+		InputDestination inputDestination = context.getBean(InputDestination.class);
+		LaunchRequest launchRequest = new LaunchRequest("test", Collections.emptyList(), Collections.emptyMap());
+		logger.info("sending:input={}", launchRequest);
+		inputDestination.send(new GenericMessage<>(launchRequest));
+		Awaitility.await("Expecting no errors")
+			.failFast(errorReceived::get)
+			.pollDelay(Duration.ofSeconds(2))
+			.timeout(Duration.ofSeconds(5))
+			.until(() -> !errorReceived.get());
 	}
 
 	@Test
-	public void launchRequestHasWrongPlatform() {
```

Comment:
This is a possible case where a user requests launch on platform A, but the task has already been launched on platform B.   In this case the launch will fail.